main.py

```diff
@@ -69,23 +86,91 @@ def draw() -> Shape:
     return target_shape
 
 
+def draw_second_stage() -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+
+    random.shuffle(shapes)
+    for index, shape in enumerate(shapes):
+        if index > 0:
+            break
+
+        position = shape.map(positions[index])
+
+        # TODO: Fix the issue with pygame and being able to draw the shape from the shape class
+        # TODO: This is a dirty hack because of the 24 hour deadline from start to finish
+        if shape.method == 'rect':
+            pygame.draw.rect(SCREEN, shape.colour, position)
+        elif shape.method == 'circle':
+            pygame.draw.circle(SCREEN, shape.colour, position, shape.radius)
+        elif shape.method == 'polygon':
+            pygame.draw.polygon(SCREEN, shape.colour, position)
+        elif shape.method == 'line':
+            pygame.draw.line(SCREEN, shape.colour, position[0], position[1], shape.radius)
+
+    target_shape = shapes[random.randint(0, 3)]
+    while target_shape.method == last_shape:
+        target_shape = shapes[random.randint(0, 3)]
+
+    font = pygame.font.Font(None, 48)
+    text = font.render(f"{current_round}. Please select the {target_shape.type}", True, (255, 255, 255))
+    text_rect = text.get_rect(center=(400, 50))
+    SCREEN.blit(text, text_rect)
+
+    return target_shape
+
+
+def draw_third_stage() -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+    path = 'D:\pedro\Documents\Tese\data_collect_py\IAPS/*.jpg'
+    IAPS = []
+    for filename in glob.glob(path):
+        IAPS.append(filename)
+    figure = random.choice(IAPS)
+    if figure in picked_IAPS:
+        while figure in picked_IAPS:
+            figure = random.choice(IAPS)
+    image = pygame.image.load(figure)
+    SCREEN.fill(0, 0, 0)
+    SCREEN.blit(image, (0, 0))
+    return figure
+
+
+def draw_whitespace():
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
```

Comment:
We're filling the screen fully black, then filling the screen fully white. We only need one of the lines, I'd suggest this method be changed to the following

```python
def clear_screen(colour: tuple = (0, 0, 0), width: int = 800, height: int = 600) -> None:
    """Draws a coloured rectangle across the entire screen, essentially clearing it of any drawings."""
    pygame.draw.rect(SCREEN, colour, (0, 0, width, height))
```

Then you can just call `clear_screen((255, 255, 255))` or something.